src/Internal/Payment/State/InitialState.php

```diff
@@ -218,6 +234,42 @@ protected function process_order_phone_number(): void {
 		}
 	}
 
+	/**
+	 * Verifies the minimum order amount, and catches the error without reaching the API.
+	 *
+	 * @throws StateTransitionException
+	 */
+	protected function verify_minimum_amount(): void {
+		$context        = $this->get_context();
+		$currency       = $context->get_currency();
+		$order_amount   = (int) $context->get_amount();
+		$minimum_amount = $this->minimum_amount_service->get_cache( $currency );
+
+		if ( $minimum_amount > $order_amount ) {
+			$error_message = $this->minimum_amount_service->get_error_message_for_shoppers( $currency, $minimum_amount );
+
+			throw new StateTransitionException( $error_message );
```

Comment:
As we've already discussed, we need to use specific exceptions rather than `StateTransitionException`s in future streams. In this case, I think it makes sense to throw the same `Amount_Too_Small_Exception` that would be thrown by the API.

Reply:
As a matter of fact, I'd say that this method can go straight to the new service, which is not aware of `StateTransitionExceptions` at all. You can pass the amount and currency to that method, and use the exception as means of communication.